frontend/src/app/modules/user-profile/user-details/user-details/user-details.component.ts

```diff
@@ -59,13 +70,25 @@ export class UserDetailsComponent extends BaseComponent implements OnInit {
                     sex: resp.sex,
                     language: resp.language,
                     languageLevel: resp.languageLevel,
+                    birthDate: resp.birthDate,
                 });
+
+                this.userService.getTagNames().pipe(this.untilThis)
+                    .subscribe(tags => this.selectedTags = tags);
             });
+
+        this.tagService.getAllTags().pipe(this.untilThis).subscribe(
+            tags => this.allTags = tags,
+        );
     }
 
     onSubmit() {
+        const userDetails = this.detailsForm.value as unknown as IUserInfo;
```

Comment:
Do you need unknown?

Reply:
yes it's needed because I couldn't cast to `IUserInfo` directly and `stackoverflow` suggested this way of solving the problem

Reply:
Please get rid of casting to unknow

Reply:
fixed